JAVASCRIPT/Task_youtube_Api/index.js

```diff
@@ -0,0 +1,254 @@
+    let searchBox = document.createElement("input");
+    searchBox.setAttribute("type", "text");
+    searchBox.setAttribute("placeholder", "Search for video");
+    searchBox.setAttribute("id", "searchBox");
+    searchBox.setAttribute("class", "searchTerm");
+    let searchButton = document.createElement("button");
+    searchButton.setAttribute("type", "submit");
+    searchButton.setAttribute("class", "searchButton");
+    searchButton.setAttribute("id", "searchButton");
+    searchButton.setAttribute("onclick", "searchBoxSubmit()")
+    let SearchIcon = document.createElement("i");
+    SearchIcon.setAttribute("class", "fa fa-search");
+    searchButton.append(SearchIcon)
+    searchDiv.append(searchBox);
+    searchDiv.append(searchButton);
+    search.append(searchDiv);
+
+
+    let epamLogo = document.createElement("div");
+    epamLogo.setAttribute("class", "li");
+    epamLogo.setAttribute("id", "epam-logo-div");
+    let epamLogoImage = document.createElement("img");
+    epamLogoImage.setAttribute("src", "./assets/img/epam-systems-logo.jpg");
+    epamLogoImage.setAttribute("class", "logo");
+    epamLogoImage.setAttribute("id", "epam-logo");
+    epamLogo.append(epamLogoImage);
+
+    navBar.append(youtubeLogo);
+    navBar.append(search);
+    navBar.append(epamLogo);
+
+    body.append(navBar);
+
+    let displayAllVid = document.createElement("div");
+    displayAllVid.setAttribute("id", "displayAllVid");
+
+    body.append(displayAllVid);
+
+    let PreviousNextButtonDiv = document.createElement("div");
+    PreviousNextButtonDiv.setAttribute("id", "prevNextBtn");
+    let previousButton = document.createElement("button");
+    previousButton.setAttribute("class", "previousButton");
+    previousButton.setAttribute("id", "previousButton");
+    previousButton.setAttribute("onclick", "previousResultSet()");
+    previousButton.append("Previous");
+    let nextButton = document.createElement("button");
+    nextButton.setAttribute("class", "nextButton");
+    nextButton.setAttribute("id", "nextButton");
+    nextButton.setAttribute("onclick", "nextResultSet()");
+    nextButton.append("Next");
+
+    PreviousNextButtonDiv.append(previousButton);
+    PreviousNextButtonDiv.append(nextButton);
+
+    body.append(PreviousNextButtonDiv);
+
+    let testShowHideBtn = document.createElement("button");
+    testShowHideBtn.setAttribute("class", "testShowHideBtn");
+    testShowHideBtn.setAttribute("id", "testShowHideBtn");
+    testShowHideBtn.append("Hide Test Results");
+    testShowHideBtn.setAttribute("onclick", "hideShowTests()");
+
+    body.append(testShowHideBtn);
+
+})();
+
+let hideShowTests = function() {
+    let flag = 1;
+
+    return function() {
+        let btn = document.getElementById("testShowHideBtn");
+        let testDiv = document.getElementsByClassName("jasmine_html-reporter")[0];
+        console.log(flag);
+        if (flag) {
+            btn.innerHTML = "Show Test Results";
+            flag = 0;
```

Comment:
instead of having flag as 0,1 , try and use flag as boolean. SInce there are only two cases it is more suitable